sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/xds/utils/RdsProtocolTransformer.java

```diff
@@ -0,0 +1,220 @@
+
+        List<String> domains = virtualHost.getDomainsList();
+
+        List<XdsRoute> xdsRoutes =
+                virtualHost.getRoutesList().stream()
+                        .map(route -> parseRoute(route))
+                        .collect(Collectors.toList());
+        xdsVirtualHost.setName(virtualHost.getName());
+        xdsVirtualHost.setRoutes(xdsRoutes);
+        xdsVirtualHost.setDomains(domains);
+        return xdsVirtualHost;
+    }
+
+    private static XdsRoute parseRoute(Route route) {
+        XdsRoute xdsRoute = new XdsRoute();
+        xdsRoute.setName(route.getName());
+        xdsRoute.setRouteMatch(parseRouteMatch(route.getMatch()));
+        xdsRoute.setRouteAction(parseRouteAction(route.getRoute()));
+        return xdsRoute;
+    }
+
+    private static XdsRouteMatch parseRouteMatch(RouteMatch routeMatch) {
+        XdsRouteMatch xdsRouteMatch = new XdsRouteMatch();
+        xdsRouteMatch.setPathMatcher(parsePathMatcher(routeMatch));
+        xdsRouteMatch.setHeaderMatchers(parseHeaderMatchers(routeMatch.getHeadersList()));
+        xdsRouteMatch.setCaseSensitive(routeMatch.getCaseSensitive().getValue());
+        return xdsRouteMatch;
+    }
+
+    private static List<XdsHeaderMatcher> parseHeaderMatchers(List<HeaderMatcher> headerMatchers) {
+        return headerMatchers.stream()
+                .filter(Objects::nonNull)
+                .map(headerMatcher -> parseHeaderMatcher(headerMatcher))
+                .collect(Collectors.toList());
+    }
+
+    private static XdsHeaderMatcher parseHeaderMatcher(HeaderMatcher headerMatcher) {
+        if (headerMatcher.getPresentMatch()) {
+            return new XdsHeaderMatcher(headerMatcher.getName(),
+                    new PresentMatchStrategy());
+        }
+        if (!headerMatcher.hasStringMatch()) {
+            LOGGER.log(Level.WARNING,
+                    "The xDS route header matching strategy is unknown. Please check the route configuration.");
+            return new XdsHeaderMatcher(headerMatcher.getName(),
+                    new UnknownMatchStrategy());
+        }
+        StringMatcher stringMatch = headerMatcher.getStringMatch();
+        switch (stringMatch.getMatchPatternCase()) {
+            case EXACT:
+                return new XdsHeaderMatcher(headerMatcher.getName(),
+                        new ExactMatchStrategy(stringMatch.getExact()));
+            case PREFIX:
+                return new XdsHeaderMatcher(headerMatcher.getName(),
+                        new PrefixMatchStrategy(stringMatch.getPrefix()));
+            case SUFFIX:
+                return new XdsHeaderMatcher(headerMatcher.getName(),
+                        new SuffixMatchStrategy(stringMatch.getSuffix()));
+            case SAFE_REGEX:
+                return new XdsHeaderMatcher(headerMatcher.getName(),
+                        new RegexMatchStrategy(stringMatch.getSafeRegex().getRegex()));
+            default:
```

Comment:
It will create only once new route configuration comes？ I hope these not being created very frequently